woocommerce/woocommerce-gutenberg-products-block
# Product Collection - New 'No Results' block with default UI

## What

This PR introduces a custom 'No Results' block for the Product Collection block, specifically designed to replace the default core 'query-no-results' block usage. This new block provides the following default view when there are no products for selected filters.

![image](https://github.com/woocommerce/woocommerce-blocks/assets/16707866/e618cce6-9fd7-4b40-bc40-0e342cc92799)


Part of #10826

## Why

Developing a new "No Results" block is essential for an improved default user interface. This initiative will lay the groundwork for enhancing the user experience for merchants. Additionally, it will facilitate the creation of extensibility options for third-party developers (3PDs). Furthermore, we plan to introduce a pattern selection feature, enabling merchants to customize the appearance of the "No Results" view. For more info, you may also check Github Discussion #10707.



## Testing Instructions



_Please consider any edge cases this change may have, and also other areas of the product this may impact._

### Case 1: Normal

1. Create a new post & add Product Collection block
2. Verify that there is "No results" block, which UI should look like this:
![image](https://github.com/woocommerce/woocommerce-blocks/assets/16707866/6a52b7d6-1323-44f5-a71c-6dee5a6e0a97)

    
3. Save the post & verify on Frontend that "No results" blocks work as expected:
    - **Note:** This block will be only visible if there are no products to show on the frontend. You can achieve this by changing the filters in Inspector control such that there are no products to show for those filters. For example, try setting "Stock Status" filter to only "On backorder".
4. There are two links on frontend:
    - **store's home:** Verify that "store's home" link takes you to the home page of the store.
    - **clearing any filters:** This functionality is tested in Case 2.

### Case 2: Testing `clearing any filters` link

1. Create a new post.
2. Add filters blocks, i.e. Filter by Price, Filter by Stock, Filter by Attribute, Filter by Rating, etc. You can use a pattern called "Product Filters" which contains all the filters.
3. Add Product Collection block & publish the post
4. Go to frontend:
    - Change all filter block values so that there are no products for selected filters and no results block becomes visible. URL will look something like this: `http://store.local/162-2/?filter_stock_status=instock&filter_color=blue%2Cgray&query_type_color=or&min_price=29&max_price=32`
    - Click on "clearing any filters" link & verify that it clears all filter values.
    - *TO BE TESTED BY DEVELOPERS*
        - You can also add some random query variables to URL, for example `&random=value` & URL will look something like this: `http://store.local/162-2/?filter_stock_status=instock&filter_color=blue%2Cgray&query_type_color=or&min_price=29&max_price=32&random=value`
            - Verify that clicking on `clearing any filters` only removes query variables that are related to filters & doesn't remove this random query variable we added.

### Case 3: Adding styles to "No results" block

1. Create a new post & add Product Collection block.
2. Select "No results" block & change styles using Inspector control, i.e.:
    - Text, Background & Link color
    - Typography: Font family, appearance, line height, decoration etc.
3. Verify the changes made to these styles are visible on Editor.
4. Publish the post & verify that changes are visible on Frontend as well.

### Case 4: Adding/removing blocks

1. Create a new post & add Product Collection block.
2. In “No results” block:
    1. Verify that you can remove existing blocks.
    2. Verify that you can add new blocks to it.
    3. Verify that the above changes are reflected correctly on the frontend.

Here is a quick demo:


https://github.com/woocommerce/woocommerce-blocks/assets/16707866/a5907a6f-8131-45f9-9c18-6f693cd02b5c





## Screenshots or screencast



| Before | After |
| ------ | ----- |
|        |       |

## WooCommerce Visibility


Required:


## Checklist

Required:


Conditional:


## Changelog


> Product Collection - New 'No Results' block with default UI
